x-pack/plugins/aiops/public/application/url_state/log_rate_analysis.ts

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { WindowParameters } from '@kbn/aiops-utils';
+
+import { getDefaultAiOpsListState, type AiOpsFullIndexBasedAppState } from './common';
+
+export interface LogRateAnalysisPageUrlState {
+  pageKey: 'logRateAnalysis';
+  pageUrlState: LogRateAnalysisAppState;
+}
+/**
+ * To avoid long urls, we store the window parameters in the url state not with
+ * their full parameters names but with abbrevations. `windowParameters2AppState` and
+ * `appState2WindowParameters` are used to transform the data structure.
+ */
+export interface LogRateAnalysisAppState extends AiOpsFullIndexBasedAppState {
+  /** Window parameters */
+  wp?: {
+    /** Baseline minimum value */
+    bMin: number;
+    /** Baseline maximum value */
+    bMax: number;
+    /** Deviation minimum value */
+    dMin: number;
+    /** Deviation maximum value */
+    dMax: number;
+  };
+}
+
+/**
+ * Transforms a full window parameters object to the abbreviated url state version.
+ */
+export const windowParameters2AppState = (wp?: WindowParameters): LogRateAnalysisAppState['wp'] =>
```

Comment:
Is it worth doing any basic validation of the parameters? For example, here I manually edited the URL so that the baseline / deviation were outside the page time range:

<img width="431" alt="image" src="https://github.com/elastic/kibana/assets/7405507/18c82d36-ee2a-454a-bcf4-3344c196e08f">

Admittedly this is an edge case, so maybe not worth the extra work?

Reply:
I documented this in a separate issue to follow up with: https://github.com/elastic/kibana/issues/171657

The problem here is that the full time range doesn't include any data, we should recognize that and not show the brushes and UI below at all since there will be no results anyway. So this can happen by updating the date picker too, it's sort of unrelated to the url state update.